### PR TITLE
Boost SM Refactor: #1

### PR DESCRIPTION
**Pull request by aarshkshah1992**
- Adding some more logging.
- Added some TODOs.
- Moved the provider event loop logic to it's own file.

### CODE REVIEW

File: storagemarket/provider.go [hunk 1]
```diff
@@ -184,9 +171,30 @@ func (p *Provider) ExecuteDeal(dp *types.ClientDealParams) (pi *api.ProviderDeal
 		return resp.ri, nil
 	}
 
+	log.Infow("scheduled deal for execution", "id", dp.DealUuid)
 	return nil, nil
 }
 
+func (p *Provider) sendDealForAcceptance(ds *types.ProviderDealState) (acceptDealResp, error) {
```
Comment (dirkmc) [inline storagemarket/provider.go:178]: nit: suggest renaming to `checkForDealAcceptance`

File: storagemarket/provider.go [hunk 2]
```diff
@@ -195,25 +203,27 @@ func (p *Provider) Start(ctx context.Context) error {
...
+		ds := deal
 		restartWg.Add(1)
 		go func() {
 			defer restartWg.Done()
 
 			select {
-			case p.restartDealsChan <- restartReq{deal: deal}:
+			case p.restartDealsChan <- restartReq{deal: ds}:
 			case <-p.ctx.Done():
+				log.Errorw("failed to restart deal", "err", p.ctx.Err(), "id", ds.DealUuid)
```
Comment (dirkmc) [inline storagemarket/provider.go:226]:
```suggestion
				log.Errorw("timeout when restarting deal", "err", p.ctx.Err(), "id", ds.DealUuid)
```

**Review by dirkmc — APPROVED** [review]
A couple of nits but LGTM 👍

File: storagemarket/provider.go (hunk 2 above)
Comment (aarshkshah1992) [reply storagemarket/provider.go]: done.

File: storagemarket/provider.go (hunk 1 above)
Comment (aarshkshah1992) [reply storagemarket/provider.go]: done.

**Pull request merged**